packages/components/src/input-control/input-base.tsx

```diff
@@ -91,20 +90,19 @@ export function InputBase(
 			{ ...props }
 			{ ...getUIFlexProps( labelPosition ) }
 			className={ className }
+			gap={ 2 }
 			isFocused={ isFocused }
 			labelPosition={ labelPosition }
 			ref={ ref }
 		>
-			<LabelWrapper>
```

Comment:
This `LabelWrapper` was moved into `Label` itself, so it can be rendered only when the label is visible. Otherwise it stays in the DOM even when the label is visually hidden, and messes with the flex gap.